test/e2e/cypress/integration/profiles/oidc-provider/login.spec.ts

```diff
@@ -146,3 +147,189 @@ context("OpenID Provider", () => {
+      expect(token.scope).to.equal("offline openid")
+      let idToken = JSON.parse(
+        decodeURIComponent(escape(window.atob(token.id_token.split(".")[1]))),
+      )
+      expect(idToken).to.have.property("amr")
+      expect(idToken.amr).to.deep.equal(amrs || ["password"])
+    })
+  }
+
+  before(() => {
+    cy.useConfigProfile("oidc-provider")
+    cy.updateConfigFile((config) => {
+      config.selfservice.allowed_return_urls = [
+        oauth2.getDefaultAuthorizeURL(client),
+      ]
+      config.oauth2_provider.return_to_enabled = true
+      return config
+    })
+    cy.proxy("express")
+  })
+
+  it("switch to registration flow", () => {
+    const identity = gen.identityWithWebsite()
+
+    const url = oauth2.getDefaultAuthorizeURL(client)
+
+    cy.visit(url)
+    cy.get("[href*='/registration']").click()
+    cy.url().should("contain", "/registration")
+
+    cy.get("[name='traits.email']").type(identity.email)
+    cy.get("[name='password']").type(identity.password)
+    cy.get("[name='traits.website']").type(identity.fields["traits.website"])
+    cy.get("[type='submit'][value='password']").click()
+
+    doConsent()
+  })
+
+  it("switch to recovery flow with password reset", () => {
+    cy.deleteMail()
+    cy.longRecoveryLifespan()
+    cy.longLinkLifespan()
+    cy.disableVerification()
+    cy.enableRecovery()
+    cy.useRecoveryStrategy("code")
+    cy.notifyUnknownRecipients("recovery", false)
+    cy.longPrivilegedSessionTime()
+
+    const identity = gen.identityWithWebsite()
+    cy.registerApi(identity)
+
+    const url = oauth2.getDefaultAuthorizeURL(client)
+    cy.visit(url)
+    cy.get("[href*='/recovery']").click()
+
+    cy.get("input[name='email']").type(identity.email)
+    cy.get("button[value='code']").click()
+    cy.get('[data-testid="ui/message/1060003"]').should(
+      "have.text",
+      "An email containing a recovery code has been sent to the email address you provided. If you have not received an email, check the spelling of the address and make sure to use the address you registered with.",
+    )
+
+    cy.recoveryEmailWithCode({ expect: { email: identity.email } })
+    cy.get("button[value='code']").click()
+
+    cy.get('[data-testid="ui/message/1060001"]', { timeout: 30000 }).should(
+      "contain.text",
+      "You successfully recovered your account. ",
+    )
+
+    cy.getSession()
+    cy.location("pathname").should("eq", "/settings")
+    // do a password change
+    const newPassword = gen.password()
+    cy.get('input[name="password"]').clear().type(newPassword)
+    cy.get('button[value="password"]').click()
+
+    // since we aren't skipping the consent, Kratos will ask us to login again
+    cy.get('input[name="password"]').clear().type(newPassword)
+    cy.get('button[value="password"]').click()
+
+    // we should now end up on the consent screen
+    doConsent(["code_recovery", "password"])
+  })
+
+  it("switch to recovery flow with oidc link", () => {
```

Comment:
this test case validates the behaviour when a user updates their profile with an OIDC provider. In this test we reuse Hydra + UI consent node app which is linked to our Kratos instance.